File: src/app/DTOs/ClientDto.ts
import { AccountStatus } from "../Enum/AccountStatus 1";
import { AccountType } from "../Enum/AccountType 1";

export interface ClientDto {
  clientId: number;
  name: string;
  address?: string | null;
  bankId: number;
  userId: number;
  accountNo: string;
  verificationStatus?: AccountStatus; 
  accountType?: AccountType;
  rejectionRemark?: string; // Add this property for rejection remarks
}

File: src/app/DTOs/ClientDto.cs
using BankingApp.Enums;

public class ClientDto
{
    public int ClientId { get; set; }
    public string Name { get; set; } = null!;
    public string? Address { get; set; }

    public int BankId { get; set; }   // required (not nullable)
    public int UserId { get; set; }   // required (not nullable)

    public string AccountNo { get; set; } = null!; // required

    public AccountStatus? VerificationStatus { get; set; } = AccountStatus.Pending; // mapped from enum
    public AccountType? AccountType { get; set; }       // mapped from enum
}